### ISSUE TITLE
Create a statement for users who entered time entries within the grace period

### ISSUE THREAD

**Issue by interlock** [lighthouse-labs/clock-tower#93]

**Comment by JBorts** [comment]
@interlock As I also understand it, if a user enters time during the grace period and no hours during the following period, then a statement won't be generated in the next period either. Is that correct?

Example:
For the Sept 16-30th period, I enter my hours on October 2nd. I then go on vacation for a month. No statement is generated on October 15th with the backdated hours either

If this is indeed the case, then isn't it fairly easy to just generate a statement if any backdated hours are open as well?

**Comment by interlock** [comment]
@JBorts Correct, basically any hours entered in the grace period for a statement automatically get associated to the statement. If there was no hours entered, at all, those hours do not get associated to a statement (since it does not exist) but instead are include in the next statement.

Statement generation for a period basically takes all hours in the period AND hours before that period. It marks those out of range hours as BACKDATED IIRC.

You are welcome to create a new rake task that runs with the grace period closing to fix it. I don't think it's trivially easy; just that no one has spent any time to do it.

**Comment by JBorts** [comment]
#137 @interlock Can you review and merge if ok?
